l10n_ec_account_edi/tests/test_l10n_ec_check_certificate.py

```diff
@@ -0,0 +1,100 @@
+import datetime
+
+from odoo import fields
+from odoo.tests import tagged
+
+from .test_edi_common import TestL10nECEdiCommon
+
+
+@tagged("post_install_l10n", "post_install", "-at_install", "certificate")
+class TestL10nCheckCertificate(TestL10nECEdiCommon):
+    def test_l10n_ec_check_certificate(self):
+        """Test that the cron task is executed correctly."""
+        self._setup_edi_company_ec()
+        self._prepare_certificates()
+
+        cron_tasks = self.env.ref(
+            "l10n_ec_account_edi.ir_cron_check_certificate", False
+        )
+        self.assertTrue(cron_tasks)
+
+        result = cron_tasks.method_direct_trigger()
+        self.assertTrue(result)
+
+        mails_send = self.env["mail.message"].search_count(
+            [
+                "|",
+                "|",
+                ("subject", "like", "%Cert 4%"),
+                ("subject", "like", "%Cert 3%"),
+                ("subject", "like", "%Cert 1%"),
+                ("model", "=", "sri.key.type"),
+                ("message_type", "=", "notification"),
+            ]
+        )
+
+        self.assertEqual(mails_send, 1)
+
+        no_mails_send = self.env["mail.message"].search_count(
+            [
+                ("subject", "like", "%Cert 2%"),
+                ("model", "=", "sri.key.type"),
+                ("message_type", "=", "notification"),
+            ]
+        )
+
+        self.assertEqual(no_mails_send, 0)
+
+        cron_tasks = self.env.ref("mail.ir_cron_mail_scheduler_action", False)
+
+        self.assertTrue(cron_tasks)
+
+    def _prepare_certificates(self):
```

Comment:
could be better this code is done on `setUpClass` instead here?

Reply:
Is fixed up